packages/itwin/tree-widget/src/components/trees/repositories-tree/RepositoriesService.ts

```diff
@@ -0,0 +1,78 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+/**
+ * @internal
+ */
+export interface ITwinRepositories {
+  repositories: ITwinRepository[];
+}
+
+interface Repository {
+  id?: string;
+  displayName?: string;
+  type?: string;
+  name?: string;
+}
+
+interface ITwinRepository {
+  id?: string;
+  class: string;
+  subClass?: string;
+  uri?: string;
+}
+
+/**
+ * @internal
+ */
+export async function getItwinRepositories(itwinId: string, accessToken: string, environment?: "PROD" | "QA" | "DEV"): Promise<ITwinRepository[]> {
+  const url = getRepositoriesUrl(itwinId, environment);
+  const result = await fetchData<ITwinRepository[]>(url, accessToken);
+  return result;
+}
+
+/**
+ * @internal
+ */
+export async function getRepositoryData(accessToken: string, url?: string): Promise<Repository[]> {
+  if (!url) return [];
+
+  const result = await fetchData<Repository[]>(url, accessToken);
+  return result;
+}
+
+async function fetchData<T>(url: string, accessToken: string): Promise<T> {
+  const headers = {
+    Authorization: accessToken,
+    "Content-Type": "application/json",
+    Prefer: "return=representation",
+  };
```

Comment:
Based on `https://developer.bentley.com/apis/iTwins/operations/get-repositories-by-itwin-id/`, it seems we should also set the `Accept` header.

Reply:
Each repository endpoint uses a different version of Accept so for now decided to not declare accept header.